web/Services/PropertyService.cs

```diff
@@ -172,23 +171,55 @@ public bool IsUserAdminOrCorrectLandlord(BricksAndHeartsUser currentUser, int pr
         return propertyLandlordId == userLandlordId;
     }
 
-    public List<PropertyDbModel> SortProperties(string? by)
+    public async Task<(List<PropertyDbModel> PropertyList, int Count)> GetPropertyList(string? sortBy, string? target, int page, int propPerPage)
     {
         List<PropertyDbModel> properties;
-        if (by == "Availability")
+        switch (sortBy)
         {
-            properties = _dbContext.Properties.OrderBy(m => m.RenterUserId).ToList();
-        }
-        else if (by == "Rent")
-        {
-            properties = _dbContext.Properties.OrderBy(m => m.Rent).ToList();
-        }
-        else
-        {
-            properties = _dbContext.Properties.ToList();
+            case "Location":
+                var postcode = _postcodeService.FormatPostcode(target!);
+                if (postcode == "")
+                {
+                    return (new List<PropertyDbModel>(), 0);
+                }
+                var postcodeList = new List<string> { postcode };
+                await _postcodeService.AddPostcodesToDatabaseIfAbsent(postcodeList);
+                var model = _dbContext.Postcodes.SingleOrDefault(p => p.Postcode == postcode);
+                if (model?.Lat == null || model.Lon == null)
+                {
+                    return (new List<PropertyDbModel>(), 0);
+                }
+
+                properties = await _dbContext.Properties
+                    .FromSqlInterpolated(
+                        @$"SELECT *, (
+                  6371 * acos (
+                  cos ( radians({model.Lat}) )
+                  * cos( radians( Lat ) )
+                  * cos( radians( Lon ) - radians({model.Lon}) )
+                  + sin ( radians({model.Lat}) )
+                  * sin( radians( Lat ) )
+                    )
+                ) AS distance 
+                FROM dbo.Property
+                WHERE Lon is not NULL and Lat is not NULL
+                ORDER BY distance
+                OFFSET {propPerPage * (page - 1)} ROWS
+                FETCH NEXT {propPerPage} ROWS ONLY

```

Comment:
if we remove these 2 lines, and make this assign to a properties IQueryable instead of a list (and then remove the Skip and Take in the other cases) we can then just add the skip and take to the end of the file and only have it written once :)

Reply:
If I do that, I get this error:
SqlException: The ORDER BY clause is invalid in views, inline functions, derived tables, subqueries, and common table expressions, unless TOP, OFFSET or FOR XML is also specified. Incorrect syntax near 'OFFSET'. Invalid usage of the option NEXT in the FETCH statement.
I'm not very familiar with the SQL commands so I'm not entirely sure what this error is, any idea what it is/how to fix it?

Reply:
I've managed to fix this by adding OFFSET 0 ROWS - so the skip and take can still appear at the end, but SQL stops having issues.